### PR TITLE
Picard: fix parsing mixed strings/numbers, account for trailing tab

### PR DESCRIPTION
**Pull request by vladsavelyev**
Fixes https://github.com/ewels/MultiQC/issues/2061, https://github.com/ewels/MultiQC/issues/2063, https://github.com/ewels/MultiQC/issues/2038

### CODE REVIEW

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 ### Module updates
 
+- **Picard**: fix parsing mixed strings/numbers, account for trailing tab ([#2083](https://github.com/ewels/MultiQC/pull/2083))
+
```
Comment (ewels) [inline CHANGELOG.md:20]: Maybe a tweak needed for the changelog bot? Updates usually go on their own newlines under module name.
```suggestion
- **Picard**:
    - Fix parsing mixed strings/numbers, account for trailing tab ([#2083](https://github.com/ewels/MultiQC/pull/2083))

```

File: multiqc/modules/picard/MarkDuplicates.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def save_table_results(s_name, base_s_name, keys, parsed_data, recompute_merged_
             #
             if "markduplicates" in l.lower() and "input" in l.lower():
                 # Pull sample name from input
-                fn_search = re.search(r"INPUT(?:=|\s+)(\[?[^\s]+\]?)", l, flags=re.IGNORECASE)
+                fn_search = re.search(r"INPUT(?:=|\s+)(\[?[^\s]+]?)", l, flags=re.IGNORECASE)
```
Comment (ewels) [inline multiqc/modules/picard/MarkDuplicates.py:93]: Why did you remove the escape character?

```suggestion
                fn_search = re.search(r"INPUT(?:=|\s+)(\[?[^\s]+\]?)", l, flags=re.IGNORECASE)
```

**Review by ewels — APPROVED** [review]

File: multiqc/modules/picard/MarkDuplicates.py (hunk 2 above)
Comment (ewels) [reply multiqc/modules/picard/MarkDuplicates.py:93]: Committing so that I can merge. Shout if you put this here intentionally.

**Comment by ewels** [comment]
@multiqc-bot fix linting

**Comment by ewels** [comment]
@multiqc-bot fix linting

**Comment by ewels** [comment]
@multiqc-bot fix linting
